sui/operators.js

```diff
@@ -0,0 +1,301 @@
+const { Command, Option } = require('commander');
+const { Transaction } = require('@mysten/sui/transactions');
+const { bcs } = require('@mysten/sui/bcs');
+const { printInfo, loadConfig } = require('../common/utils');
+const { operatorsStruct } = require('./types-utils');
+const { addBaseOptions, addOptionsToCommands, parseSuiUnitAmount } = require('./cli-utils');
+const { getWallet, printWalletInfo, broadcast } = require('./sign-utils');
+const { getBcsBytesByObjectId, findOwnedObjectId } = require('./utils');
+
+async function getGasCollectorCapId(client, gasServiceConfig, operatorsConfig) {
+    const operatorId = operatorsConfig.objects.Operators;
+
+    // Get and parse operator data
+    const operatorBytes = await getBcsBytesByObjectId(client, operatorId);
+    const parsedOperator = operatorsStruct.parse(operatorBytes);
+
+    // Get the capabilities bag ID
+    const bagId = parsedOperator.caps.id;
+
+    // Find the GasCollectorCap bag ID
+    const bagResult = await client.getDynamicFields({
+        parentId: bagId,
+        name: 'caps',
+    });
+    const gasCollectorBagId = bagResult.data.find(
+        (cap) => cap.objectType === `${gasServiceConfig.address}::gas_service::GasCollectorCap`,
+    )?.objectId;
+
+    if (!gasCollectorBagId) {
+        throw new Error('GasCollectorCap not found in the operator capabilities bag');
+    }
+
+    // Get the actual cap ID from the bag ID
+    const gasCollectorCapObject = await client.getObject({
+        id: gasCollectorBagId,
+        options: {
+            showContent: true,
+        },
+    });
+
+    // Extract and return the gas collector cap ID
+    const gasCollectorCapId = gasCollectorCapObject.data.content.fields.value.fields.id.id;
```

Comment:
hmmmm, isn't there a structured helper to retrieve bag items?

Reply:
do you mean in the sui sdk? no, i haven't seen any utils for this.

let me create one for it

Reply:
i see, that's weird. Can you ask in the mysten channel if there's a shortcut/conventional way to do this?

Reply:
I've asked him in the `mysten` channel. Here's Damir answer:

```
Frontend wise, while we still haven’t launched GraphQL fully, most of the solutions won’t be optimal. You’ll have to query the object, find the bag id, and then do either a direct request of a field, or query all dynamic fields and use this information to query the ones you need.

When not using dynamic fields, if you want to avoid long property chains with fields.xxx, you can add the `showBcs: true”` and then parse bcs bytes “manually”. I’ll be able to share an example of this later today.
```

Reply:
I've moved this into a util function